src/Calculator/Views/StateTriggers/ControlSizeTrigger.cpp

```diff
@@ -51,30 +53,19 @@ void ControlSizeTrigger::UnregisterSizeChanged(FrameworkElement^ element)
     }
 }
 
-void ControlSizeTrigger::OnSizeChanged(Object^ sender, SizeChangedEventArgs^ e)
+void ControlSizeTrigger::OnSizeChanged(Object ^ sender, SizeChangedEventArgs ^ e)
 {
     UpdateIsActive(e->NewSize);
 }
 
 void ControlSizeTrigger::UpdateIsActive(Size sourceSize)
 {
-    if (Source != nullptr && MinWidth > 0 || MinHeight > 0)
+    if (MinHeight >= 0)
     {
-        if (MinHeight > 0 && MinWidth > 0)
-        {
-            SetActive((sourceSize.Height >= MinHeight) && (sourceSize.Width >= MinWidth));
-        }
-        else if (MinHeight > 0)
-        {
-            SetActive(sourceSize.Height >= MinHeight);
-        }
-        else
-        {
-            SetActive(sourceSize.Width >= MinWidth);
-        }
+        SetActive(sourceSize.Height >= MinHeight && (MinWidth < 0 || sourceSize.Width >= MinWidth));
     }
     else
     {
-        SetActive(false);
+        SetActive(MinWidth >= 0 && sourceSize.Width >= MinWidth);
```

Comment:
We can remove `MinWidth>=0` because sourceSize _should_ always be >= 0, but I prefer to keep it to be sure this assumption is and will always be true